tests/test_embed/test_interpreter.cpp

```diff
@@ -282,3 +282,55 @@ TEST_CASE("Reload module from file") {
     result = module.attr("test")().cast<int>();
     REQUIRE(result == 2);
 }
+
+namespace rerun_register_exception
```

Comment:
Very minor detail: do we need this namespace here? Why did you add it?
It might be good practice, but it's also not used for the other tests, so maybe consistency is more important?